### PR TITLE
databases-week2

### PR DESCRIPTION
**Pull request by somayehdvr**

### CODE REVIEW

File: databases/week2/hw_part1.sql [hunk 1]
```diff
@@ -0,0 +1,47 @@
+-- Part 1: Working with tasks
+-- Write the following sql queries:
+-- In all the queries, you choose which is the task that you will modify/delete.
+-- Add a task with these attributes: 
+-- title, description, created, updated, due_date, status_id, user_id
+INSERT INTO 
+    task(title, description, created, updated, due_date, status_id, user_id)
+VALUES
+    ('Do the homework', 'You have DB homework to do', '2022-09-20 19:34:43', '2022-10-15 23:35:45', '2022-12-24 16:00:46', 2, 5);
+
```
Comment (Elia-Git) [inline databases/week2/hw_part1.sql:10]: Do we have column `user_id` in the table task as yet? What results gave you this query?
You can create column user_id into table task as below
```
ALTER TABLE   task 
ADD   user_id int(10) unsigned NOT NULL;
```

File: databases/week2/hw_part1.sql [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+SET
+    status_id = 3
+WHERE
+    id = 35;
+
+-- Delete a task
+DELETE FROM
+    task
+WHERE
+    id = 36;
```
Comment (Elia-Git) [inline databases/week2/hw_part1.sql:47]: Good job here even though the task was not created on the above query!

File: databases/week2/hw_part2.sql [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+-- If you are done with the above tasks, you can continue with these advanced tasks:
+-- Create an index on the name column of the student table.
+CREATE INDEX `name_index`
+ON `student`(`name`);
+
+-- Add a new column to the class table named status which can only have the 
+-- following values: not-started, ongoing, finished (hint: enumerations).
+ALTER TABLE `class`
+ADD `status` ENUM('not-started', 'ongoing', 'finished');
```
Comment (Elia-Git) [inline databases/week2/hw_part2.sql:33]: Make full sense, thanks!

File: databases/week2/hw_part4.sql [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+);
+CREATE TABLE `order_datails`(
+    `id` INT(10) UNSIGNED NOT NULL AUTO_INCREMENT,
+    `order_id` INT(10) UNSIGNED NOT NULL,
+    `product_id` INT(10) UNSIGNED NOT NULL,
+    `quantity` INT(10) UNSIGNED NOT NULL,
+    PRIMARY KEY (`id`),
+    CONSTRAINT `fk_order` FOREIGN KEY (`order_id`) REFERENCES `order` (`id`) ON DELETE CASCADE ON UPDATE CASCADE,
+    CONSTRAINT `fk_product` FOREIGN KEY (`product_id`) REFERENCES `product` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
+);
```
Comment (Elia-Git) [inline databases/week2/hw_part4.sql:45]: Nice job. Do have an ED diagram for this?

File: databases/week2/hw_part3.sql [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+-- Get all the tasks for 'Maryrose Meadows' that were created in september (hint: month(created)=month_number)
+SELECT 
+    task.*
+FROM
+    task
+    LEFT JOIN user ON task.user_id = user.id
+WHERE
+    user.name = 'Maryrose Meadows'
+    AND month(task.created) = 9;
```
Comment (Elia-Git) [inline databases/week2/hw_part3.sql:31]: This queries when column user_id introduced to table task !

**Review by Elia-Git — COMMENTED** [review]
Good job in general. Just a few comments for a small fix.

File: databases/week2/hw_part4.sql (hunk 4 above)
Comment (somayehdvr) [reply databases/week2/hw_part4.sql:45]: yes it's erd.drawio.png file.

File: databases/week2/hw_part1.sql (hunk 1 above)
Comment (somayehdvr) [reply databases/week2/hw_part1.sql:10]: we have this column in the database already. In the lesson-data.sql it's been created.

**Pull request merged**